client/web/compose/src/components/PageBlocks/Navigation/Base.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+  <wrap
+    v-bind="$props"
+    v-on="$listeners"
+  >
+    <div class="h-100 w-100">
+      <b-nav
+        :tabs="options.display.appearance === 'tabs'"
+        :pills="options.display.appearance === 'pills'"
+        :small="options.display.appearance === 'small'"
+        :fill="options.display.fillJustify === 'fill'"
+        :justify="options.display.fillJustify === 'justify'"
+        :align="options.display.alignment"
+        class="border-0 h-100 navigation-page-block-item"
+      >
+        <component
+          :is="displayNavItemDrop(list)"
+          v-for="(list, index) in options.lists"
+          :key="`lists-${index}`"
+          :to="list.options.item.pageID"
+          :href="list.options.item.url"
+          :disabled="list.options.disabled"
+          :variant="list.options.backgroundColor"
+          :style="{ order: index, color: list.options.textColor }"
+          :link-attrs="{ style: `color: ${list.options.textColor}` }"
+          :target="selectTargetOption(list.options.item.target)"
+          :text="list.options.item.label"
+          :right="list.options.align === 'right'"
+          :left="list.options.align === 'left'"
+          class="nav-item-dropdown"
+        >
+          <template v-if="!list.options.item.displaySubPages && list.type !== 'dropdown'">
+            <span>
+              {{ list.options.item.label }}
+            </span>
+          </template>
+
+          <template v-if="(list.type === 'dropdown')">
+            <b-dropdown-item
+              v-for="(dropdown, dIndex) in list.options.item.dropdown"
+              :key="`dropdown-${dIndex}`"
+              :to="dropdown.url"
+              :disabled="list.options.disabled"
+              :style="{ order: dIndex * 2 }"
+            >
+              {{ dropdown.label }}
+            </b-dropdown-item>
+
+            <b-dropdown-divider
+              v-for="(dropdown, dIndex) in list.options.dropdown"
+              v-show="dropdown.delimiter"
+              :key="`dropdown-divider-${dIndex}`"
+              :style="{ order: dIndex + 1 }"
+            />
+          </template>
+
+          <template v-if="(list.type === 'compose' && list.options.item.displaySubPages)">
+            <b-dropdown-item
+              v-for="(dropdown, dIndex) in getSubPages(list.options.item.pageID)"
+              :key="`dropdown-${dIndex}`"
+              :disabled="list.options.disabled"
+              :to="dropdown.pageID"
+            >
+              {{ dropdown.title }}
+            </b-dropdown-item>
+          </template>
+        </component>
+      </b-nav>
+    </div>
+  </wrap>
+</template>
+<script>
+import { mapGetters } from 'vuex'
+import base from '../base'
+
+export default {
+  extends: base,
+
+  computed: {
+    ...mapGetters({
+      pages: 'page/set',
+    }),
+  },
+
+  methods: {
+    displayNavItemDrop (list) {
+      if (list.options.item.label && (!list.options.item.displaySubPages && list.type !== 'dropdown')) {
```

Comment:
If you think about it you can improve the conditions here a bit. since two of the conditions share a variable.
list type !== 'dropdown' and list.type === 'dropdown', also the displaySubPages is used in two conditions, so else-if might come in handy. The same for template ifs (line 32-66)